src/types/Person.ts

```diff
@@ -6,6 +6,9 @@ export interface Person {
   fatherName: string | null;
   motherName: string | null;
   slug: string;
-  mother?: Person;
-  father?: Person;
+}
```

Comment:
Interesting solution, though not so necessary.  You could use a simple Person type too, and create some function (in utils folder for example) to define whether the person has any of their parents on the current list.